pkg/i2gw/providers/gce/gateway_converter_test.go

```diff
@@ -287,6 +400,132 @@ func Test_irToGateway(t *testing.T) {
 			},
 			expectedErrors: field.ErrorList{},
 		},
+		{
+			name: "ingress with a Backend Config specifying custom HTTP health check",
+			ir: intermediate.IR{
+				Gateways: map[types.NamespacedName]intermediate.GatewayContext{
+					{Namespace: testNamespace, Name: testGatewayName}: {
+						Gateway: testGateway,
+					},
+				},
+				HTTPRoutes: map[types.NamespacedName]intermediate.HTTPRouteContext{
+					{Namespace: testNamespace, Name: testHTTPRouteName}: {
+						HTTPRoute: testHTTPRoute,
+					},
+				},
+				Services: map[types.NamespacedName]intermediate.ProviderSpecificServiceIR{
+					{Namespace: testNamespace, Name: testServiceName}: {
+						Gce: &intermediate.GceServiceIR{
+							HealthCheck: &intermediate.HealthCheckConfig{
+								CheckIntervalSec:   common.PtrTo(testCheckIntervalSec),
+								TimeoutSec:         common.PtrTo(testTimeoutSec),
+								HealthyThreshold:   common.PtrTo(testHealthyThreshold),
+								UnhealthyThreshold: common.PtrTo(testUnhealthyThreshold),
+								Type:               common.PtrTo(protocolHTTP),
+								Port:               common.PtrTo(testPort),
+								RequestPath:        common.PtrTo(testRequestPath),
+							},
+						},
+					},
+				},
+			},
+			expectedGatewayResources: i2gw.GatewayResources{
+				Gateways: map[types.NamespacedName]gatewayv1.Gateway{
+					{Namespace: testNamespace, Name: testGatewayName}: testGateway,
+				},
+				HTTPRoutes: map[types.NamespacedName]gatewayv1.HTTPRoute{
+					{Namespace: testNamespace, Name: testHTTPRouteName}: testHTTPRoute,
+				},
+				GatewayExtensions: []unstructured.Unstructured{
+					*testHealthCheckPolicyHTTPUnstructured,
```

Comment:
Especially if we have a function that generates these health check policies, would recommend just doing that inline here if possible so understanding this test doesn't require quite as much scrolling around.

Reply:
Updated.